### PR TITLE
Setup Travis CI

### PR DESCRIPTION
**Pull request by CamStan**
This sets up Travis CI testing and builds on multiple compilers that are on our machines. Can always scale back if it's overkill.

This will allow PR #4 to be closed since @becker33 has abandoned us.

### CODE REVIEW

**Comment by tonyhutter** [comment]
I love the multiple compiler targets!  Awesome!

1. Do you need `-DMPI=ON`?:
https://travis-ci.com/ECP-VeloC/AXL/jobs/187585322#L795

2. Can you have the builds fail on warnings?:
https://travis-ci.com/ECP-VeloC/AXL/jobs/187585322#L875

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  - mkdir $HOME/KVTree/kvtree-build
+  - mkdir $HOME/KVTree/kvtree-install
+  - cd $HOME/KVTree/kvtree-build
+  - cmake .. -DCMAKE_C_COMPILER=$CC -DCMAKE_CXX_COMPILER=$CXX -DCMAKE_INSTALL_PREFIX=$HOME/KVTree/kvtree-install -DMPI=ON
+  - make
+  - make install
+
+script:
+  - cd $TRAVIS_BUILD_DIR
+  - cmake -DCMAKE_C_COMPILER=$CC -DCMAKE_CXX_COMPILER=$CXX -DCMAKE_C_FLAGS="-coverage" -DMPI=ON -DAXL_ASYNC_DAEMON=OFF -DWITH_KVTREE_PREFIX=$HOME/KVTree/kvtree-install
```
Comment (CamStan) [inline .travis.yml]: Hmm, I do have `-DMPI=ON` but
```
CMake Warning:
  Manually-specified variables were not used by the project:

    MPI
```
still shows up. This shows up when I build locally as well, but only for AXL.

Comment (tonyhutter) [reply .travis.yml]: Yea, I think we can get rid of it

**Comment by CamStan** [comment]
I've never had CI set up to fail on warnings, but it might be possible if we set the `-Werror` C_FLAG. I'd have to test that if that's the route we want to go.

**Comment by tonyhutter** [comment]
My vote would be for `-Wall -Werror`

**Comment by adammoody** [comment]
I don't think AXL has an MPI option, since it doesn't use MPI.  Also, we probably don't need to enable MPI in the kvtree build for AXL either if you want to leave that out.

**Comment by CamStan** [comment]
Pulled out MPI since AXL doesn't use it and KVTree's Travis already tests with it.

I added `-Wall -Werror` and started cleaning up the errors to get it to pass, but ran into an issue here https://travis-ci.org/ECP-VeloC/AXL/jobs/511226558#L723 where Travis complains the function is unused.

Essentially, `axl_async_daemon()` only gets called by other functions inside of `#ifdef HAVE_DAEMON` blocks which in this case is turned off.

https://github.com/ECP-VeloC/AXL/blob/d427c238d0f24dd874a7deb852f219dc79695376/src/axl_async_daemon.c#L335-L337

Any thoughts?

**Comment by CamStan** [comment]
Or would it be better to leave `-Wall -Werror` out for now so that we can get Travis at least building all PR's? Then we can fix those warnings in other PR's and then put those flags back in?

**Comment by tonyhutter** [comment]
> Or would it be better to leave -Wall -Werror out for now so that we can get Travis at least building all PR's? 

I'm fine with that

**Comment by CamStan** [comment]
Issue #31 addresses the warnings for turning on `-Werror`.

Otherwise this should be good to go now.

**Comment by adammoody** [comment]
Nice!  Thanks @CamStan and @tonyhutter .

**Pull request merged**